### PR TITLE
Sanitize url_for arguments before they are passed

### PR DESCRIPTION
**Pull request by potiuk**
The url_for of flask has special arguments that start with `_` and we should sanitize the ones that come with the request before passing them.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -154,6 +154,18 @@ def truncate_task_duration(task_duration):
...
+def sanitize_args(args: dict[str, str]) -> dict[str, str]:
+    """
+    Remove all parameters starting with `_`
+    :param args: arguments of request
+    :return: the dictionary passed as input with args starting with `_` removed.
+    """
+    for key in list(args.keys()):
+        if key.startswith("_"):
+            del args[key]
+    return args
```
Comment (ashb) [inline airflow/www/views.py]: This mutates args in place which might be unexpected/cause some unforeseen issues down the line so:

```suggestion
    return {
        key: value
        for key, value in args.items()
        if not key.startswith("_")
    }
```

File: airflow/www/views.py [hunk 2]
```diff
@@ -154,6 +154,18 @@ def truncate_task_duration(task_duration):
     return int(task_duration) if task_duration > 10.0 else round(task_duration, 3)
 
 
+def sanitize_args(args: dict[str, str]) -> dict[str, str]:
+    """
+    Remove all parameters starting with `_`
```
Comment (ashb) [inline airflow/www/views.py:159]:
```suggestion
    Remove all parameters starting with `_`
    
```

Need a blank line for it to render properly

File: airflow/www/views.py (hunk 1 above)
Comment (potiuk) [reply airflow/www/views.py]: Yeah. I was pondering on that one too.. Will change.

**Review by ashb — APPROVED** [review]

File: airflow/www/views.py (hunk 2 above)
Comment (potiuk) [reply airflow/www/views.py:159]: yep

**Pull request merged**